src/main/java/de/fraunhofer/aisec/cpg/graph/Node.java

```diff
@@ -259,6 +267,15 @@ public boolean isImplicit() {
     return this.implicit;
   }
 
+  @NonNull
+  public List<Annotation> getAnnotations() {
+    return annotations;
+  }
+
+  public void setAnnotations(@NonNull List<Annotation> annotations) {
```

Comment:
The only issue that I have with this, is that we create two list for each node, if `processAnnotations` is enabled. Once for each node in the init and another one in the `CXXLanguageFrontend::handleAttributes` function which is then used in this setter here. We should probably follow a similar pattern like the DFG and EOG properties that *add* to it (or clear it) rather than creating a new list.

Reply:
Agree, re-creating the collections is not expensive, but we should keep it consistent and possibly even make collection fields `final`  one day.